### PR TITLE
Show context menu after 500ms mouse hover

### PR DESCRIPTION
**Pull request by lukebarnard1**
This implements https://github.com/vector-im/riot-web/issues/3439

There's a bit of extra faffing to make the `RoomTileContextMenu` disappear `onMouseLeave`, which seemed better than having it accidentally appearing and needing an extra click to get rid of it.

### CODE REVIEW

File: src/components/views/rooms/RoomTile.js [hunk 1]
```diff
@@ -126,36 +135,39 @@ module.exports = React.createClass({
...
+        const x = elementRect.right + window.pageXOffset + 3;
+        const chevronOffset = 12;
+        let y = (elementRect.top + (elementRect.height / 2) + window.pageYOffset);
+        y = y - (chevronOffset + 8); // where 8 is half the height of the chevron
+
+        ContextualMenu.createMenu(RoomTileContextMenu, {
+            chevronOffset: chevronOffset,
+            left: x,
+            top: y,
+            closeOnMouseLeave: true,
```
Comment (dbkr) [inline src/components/views/rooms/RoomTile.js]: Is this meant to be false if the menu was opened with a click or something? It seems to be an option that's only ever set to true.

Comment (lukebarnard1) [inline src/components/views/rooms/RoomTile.js]: It's not set to true for other `ContextualMenu`s

**Comment by lukebarnard1** [comment]
For me:
 - Get rid of `ContextMenu` background optionaly
 - Close context menu from room tile when `ev.target === ev.currentTarget`

**Comment by lukebarnard1** [comment]
I shall suggest shelving this for now as it's becoming more and more complex and is probably not worth prioritising.

**Comment by lukebarnard1** [comment]
Closing this for now

**Pull request closed without merging**